servus/serializable.cpp

```diff
@@ -23,37 +23,128 @@
 
 namespace servus
 {
+
+class Serializable::Impl
+{
+public:
+    void registerChangedCallback( const Serializable::ChangeFunc& func )
+    {
+        changed = func;
+    }
+
+    void registerDeserializedCallback(
+                                     const Serializable::DeserializeFunc& func )
+    {
+        deserialized = func;
+    }
+
+    void registerSerializeCallback( const Serializable::SerializeFunc& func )
+    {
+        serialize = func;
+    }
```

Comment:
The idea here is that when we need more than one observer and start using signals, we can keep the same interface for the connection of the signals/slots. That way the signals remain hidden in the implementation, and if we use boost, for example, the dependency doesn't leak through the header.